### ISSUE TITLE
Solution meta-project doesn't propagate target return values

### ISSUE THREAD

**Issue by tmat** [Microsoft/msbuild#3658]
### Steps to reproduce

Project file a.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net46</TargetFramework>
  </PropertyGroup>

  <Target Name="T" Returns="@(Thing)">
    <ItemGroup>
      <Thing Include="A"/>
      <Thing Include="B"/>
      <Thing Include="C"/>
    </ItemGroup>
  </Target>
</Project>
```

Directory contents:
```
/
- a.cs
- b.cs
- dir/
     - c.cs
```

Command line
```
msbuild /bl
```
### Environment data
`msbuild /version` output:

OS info:

If applicable, version of the tool that invokes MSBuild (Visual Studio, dotnet CLI, etc):

**Comment by rainersigwald** [comment]
I think that'd be a reasonable enhancement, though I hope it's never needed.

**Comment by tmat** [comment]
Why should it not be needed? It if was not needed I wouldn't file an issue. 

I would like to collect some information from all projects in a given solution. For example, find all test projects and all platforms they target and need to run tests on, so we can schedule test runs more efficiently than if we launch XUnit from within each test project.

**Comment by rainersigwald** [comment]
For that use, consider waiting for https://github.com/Microsoft/vstest/issues/411, which the test folks are working on.

Calling targets for every project in a solution is error-prone, because solutions can contain non-MSBuild projects and heterogeneous projects that don't all contain the desired target. So building infrastructure that depends on it seems risky to me.

**Comment by tmat** [comment]
> because solutions can contain non-MSBuild projects and heterogeneous projects that don't all contain the desired target

Good point. It would also be useful to have an option for skipping non-existing targets, similarly to the option `SkipNonexistentProjects`.

**Comment by rainersigwald** [comment]
There is `SkipNonexistentTargets` as of https://github.com/Microsoft/msbuild/pull/2541. But it's not clear to me that always specifying it is the right thing to do for `msbuild a.sln /t:Custom`. For `Test` you'd want it, but for a hypthetical `ValidateProjectDefinesSomeProperty` you wouldn't. We don't currently specify it for the generated traversal targets.

**Comment by tmat** [comment]
@rainersigwald Cool. Yes, it's definitely dependent on the scenario. You can either require all projects in the solution to import targets file with the custom target and then expect it to be everywhere or not if it's optional.
